include/ensmallen_bits/smorms3/smorms3_update.hpp

```diff
@@ -94,26 +100,24 @@ class SMORMS3Update
       g2 = (1 - r) % g2;
       g2 += r % (gradient % gradient);
 
-      MatType x = (g % g) / (g2 + parent.epsilon);
-
-      x.transform( [stepSize](typename MatType::elem_type &v)
-          { return std::min(v, (typename MatType::elem_type) stepSize); } );
-
-      iterate -= gradient % x / (arma::sqrt(g2) + parent.epsilon);
+      MatType x = min((g % g) / (g2 + parent.epsilon), lr);
```

Comment:
I think that using `clamp()` would actually be easier here, I guess `x = clamp(..., 0, lr)`.  (I think `x` is always nonnegative... but I might be wrong on that, I'm not sure.  Double check me on that one...)